modules/rubiconBidAdapter.js

```diff
@@ -591,43 +591,54 @@ export function hasVideoMediaType(bidRequest) {
  * @returns {string|undefined} Returns 'video' or 'banner' if resolves to a type, or undefined otherwise (invalid).
  */
 function bidType(bid, log = false) {
-  let validVideo;
+  // Is it considered video ad unit by rubicon
   if (hasVideoMediaType(bid)) {
-    validVideo = true;
-
-    if (utils.deepAccess(bid, `mediaTypes.${VIDEO}.context`) === 'instream' || bid.mediaType === VIDEO) {
-      if (typeof utils.deepAccess(bid, 'params.video.size_id') === 'undefined') {
+    // legacy mediaType or the new mediaTypes
+    // this is the preffered "new" way to define mediaTypes
+    if (typeof utils.deepAccess(bid, `mediaTypes.${VIDEO}`) !== 'undefined') {
+      // We require either context as instream or outstream
+      if (['outstream', 'instream'].indexOf(utils.deepAccess(bid, `mediaTypes.${VIDEO}.context`)) === -1) {
         if (log) {
-          utils.logError('Rubicon bid adapter Error: size id is missing for instream video request.');
+          utils.logError('Rubicon bid adapter requires mediaTypes.video.context to be one of outstream or instream');
         }
-        validVideo = false;
+        return undefined;
```

Comment:
Line 604: `return undefined;` can be shortened to just `return;`, since calling return without a value returns `undefined` as default.